src/main/java/org/jitsi_modified/impl/neomedia/rtp/TransportCCEngine.java

```diff
@@ -189,30 +185,19 @@ public void tccReceived(RtcpFbTccPacket tccPacket)
 
             if (packetDetail == null)
             {
-                return;
+                continue;
             }
-            long delta = recvTimestamp - tccPacket.getReferenceTimeMs();
-//            logger.info("Got tcc for packet " + tccSeqNum + ", the reference time is " + tccPacket.getFci().getReferenceTimeMs() +
-//                    " and it was received by the far side at " + recvTimestamp + ", meaning it has a delta of " +
-//                            delta + ".  it was originally sent at " +
-//                            packetDetail.packetSendTimeMs + " meaning in that clock it arrived at " +
-//                            (packetDetail.packetSendTimeMs + delta));
-            long arrivalTimeMs = packetDetail.packetSendTimeMs + delta;
-
-//            logger.info("Notifying bitrate estimator of incoming packet info: " +
-//                    "arrival time: " + arrivalTimeMs + ", sendTime24Bits: " + sendTime24bits +
-//                    " packet length: " + packetDetail.packetLength + ", ssrc: " +
-//                    tccPacket.getMediaSourceSsrc());
+
+            long arrivalTimeMsInLocalClock = (long)currArrivalTimestampMs - remoteReferenceTimeMs + localReferenceTimeMs;
```

Comment:
I wonder how much precision we're losing here. We could use the original resolution of `currArrivalTimestamp` and `removeReferenceTime` here. Actually, the resolution of the abs-send-time format is less than a millisecond, so we should calculate the below using the original `us` as well. 
We should also make sure we do rounding when we use integer division (we do a /1000 above)

Reply:
just so you know, we never actually track anything in actual microseconds: anywhere microseconds are used they were derived from a millisecond-precision timestamp

Reply:
Doesn't this handle tcc feedback received from the network?

Reply:
what do you mean?